test/node/long.test.ts

```diff
@@ -21,4 +22,53 @@ describe('Long', function () {
     expect(new Long(13835058055282163712n).toString()).to.equal('-4611686018427387904');
     expect(new Long(13835058055282163712n, true).toString()).to.equal('13835058055282163712');
   });
+
+  describe('static fromExtendedJSON()', function () {
+    it('is not affected by the legacy flag', function () {
+      const ejsonDoc = { $numberLong: '123456789123456789' };
+      const longRelaxedLegacy = Long.fromExtendedJSON(ejsonDoc, { legacy: true, relaxed: true });
+      const longRelaxedNonLegacy = Long.fromExtendedJSON(ejsonDoc, {
+        legacy: false,
+        relaxed: true
+      });
+      const longCanonicalLegacy = Long.fromExtendedJSON(ejsonDoc, { legacy: true, relaxed: false });
+      const longCanonicalNonLegacy = Long.fromExtendedJSON(ejsonDoc, {
+        legacy: false,
+        relaxed: false
+      });
+
+      expect(longRelaxedLegacy).to.deep.equal(longRelaxedNonLegacy);
+      expect(longCanonicalLegacy).to.deep.equal(longCanonicalNonLegacy);
+    });
+
+    describe('rejects with BSONError', function () {
+      it('hex strings', function () {
+        const ejsonDoc = { $numberLong: '0xffffffff' };
+        expect(() => Long.fromExtendedJSON(ejsonDoc)).to.throw(BSONError);
+      });
+
+      it('octal strings', function () {
+        const ejsonDoc = { $numberLong: '0o1234567' };
+        expect(() => Long.fromExtendedJSON(ejsonDoc)).to.throw(BSONError);
+      });
+
+      it('strings longer than 22 characters', function () {
+        const ejsonDoc = { $numberLong: '99999999999999999999999' };
+        expect(() => Long.fromExtendedJSON(ejsonDoc)).to.throw(BSONError);
+      });
```

Comment:
And we still want to get to a conclusion on checking if the string is 22 characters but is greater than / less than int64.max and min, respectively, we can add a tests with those input values here regardless